lib/api/client-commands/cookies/get.js

```diff
@@ -0,0 +1,76 @@
+const ClientCommand = require('../_base-command.js');
+
+/**
+ * Retrieve a single cookie visible to the current page.
+ *
+ * The cookie is returned as a cookie JSON object, with properties as defined [here](https://www.w3.org/TR/webdriver/#dfn-table-for-cookie-conversion).
+ *
+ * @example
+ * module.exports = {
+ *   'get a cookie': function (browser) {
+ *     browser
+ *       .cookies.get('test_cookie', function (result) {
+ *         this.assert.equal(result.name, 'test_cookie');
+ *         this.assert.equal(result.value, '123456');
+ *       });
+ *   },
+ *
+ *   'get a cookie with ES6 async/await': async function (browser) {
+ *     const cookie = await browser.cookies.get('test_cookie');
+ *     browser.assert.equal(cookie.name, 'test_cookie');
+ *     browser.assert.equal(cookie.value, '123456');
+ *   }
+ * };
+ *
+ * @syntax .cookies.get(name, [callback])
+ * @method cookies.get
+ * @param {string} name The cookie name.
+ * @param {function} [callback] Callback function which is called with the result value.
+ * @returns {object|null} The cookie object with properties as defined [here](https://www.w3.org/TR/webdriver/#dfn-table-for-cookie-conversion), or `null` if the cookie wasn't found.
+ * @see cookies.getAll
+ * @api protocol.cookies
+ */
+class GetCookie extends ClientCommand {
+  get returnsFullResultObject() {
+    return false;
+  }
+
+  get resolvesWithFullResultObject() {
+    return false;
+  }
+
+  /**
+   * Perform the .cookie() protocol action and pass the result to the supplied callback
+   * with the original "this" context
+   *
+   * @param {function} actionCallback
+   */
+  performAction(actionCallback) {
+    const {cookieName} = this;
+
+    this.api.cookie('GET', function(result) {
```

Comment:
here, and everywhere else in similar situations, we need to use the transport mappings directly to avoid the overhead from calling the commands via the api queue. So in v3 there should not be any commands calling other api commands via the queue.

Reply:
Done.